From 43da21e2eedf5a9c6781dfc2d199802ea1572fd5 Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan@gerhold.net>
Date: Wed, 6 Jan 2021 12:46:06 +0100
Subject: [PATCH 114/210] drm/panel: motorola-harpia-tianma: Swab brightness
 value

Seems like this panel expects the brightness value to be big endian
instead of the little endian used by mipi_dsi_dcs_set_display_brightness().

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
---
 .../drm/panel/msm8916-generated/panel-motorola-harpia-tianma.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/msm8916-generated/panel-motorola-harpia-tianma.c b/drivers/gpu/drm/panel/msm8916-generated/panel-motorola-harpia-tianma.c
index 785488885b7..6d63cf3aecc 100644
--- a/drivers/gpu/drm/panel/msm8916-generated/panel-motorola-harpia-tianma.c
+++ b/drivers/gpu/drm/panel/msm8916-generated/panel-motorola-harpia-tianma.c
@@ -205,7 +205,8 @@ static int tianma_499_v2_bl_update_status(struct backlight_device *bl)
 
 	dsi->mode_flags &= ~MIPI_DSI_MODE_LPM;
 
-	ret = mipi_dsi_dcs_set_display_brightness(dsi, brightness);
+	// This panel needs the high and low bytes swapped for the brightness value
+	ret = mipi_dsi_dcs_set_display_brightness(dsi, swab16(brightness));
 	if (ret < 0)
 		return ret;
 
-- 
2.34.1

